### PR TITLE
Add types to `app-wrapper.js`

### PR DESCRIPTION
**Pull request by lishaduck**
I'm unsure as what type `flags` is supposed to be, beyond that it's not `Flags`.
I'm pretty sure VSCode was wrong that it's `Options`.

_**Relates to:** #125_

### CODE REVIEW

File: lib/app-wrapper.js [hunk 1]
```diff
@@ -2,12 +2,25 @@ const path = require('path');
 const ElmCommunication = require('./elm-communication');
 const loadCompiledElmApp = require('./load-compiled-app');
 
+/**
+ * @typedef { import('./types/options').Options } Options
+ * @typedef { import('./types/options').ReviewOptions } ReviewOptions
+ * @typedef { import('./types/app').App } App
+ */
```
Comment (lishaduck) [inline lib/app-wrapper.js:9]: Once ts 5.5 is stable (which should be soon), we can switch to the import syntax instead here:
```suggestion
/**
 * @import {Options} from './types/options';
 * @import {ReviewOptions} from './types/options';
 * @import {App} from './types/app';
 */
```

**Comment by lishaduck** [comment]
Oh, I see. This is blocked on #126. I'll try to see if I fix it sometime.

**Comment by lishaduck** [comment]
Ok, the failures from #126 were a bit intimidating, so I figure I'll just port them over here and just coauthor them.

**Comment by lishaduck** [comment]
I plan on relanding this as separate PRs later this week.

**Comment by jfmengels** [comment]
> I'm unsure as what type flags is supposed to be, beyond that it's not Flags.
I'm pretty sure VSCode was wrong that it's Options.

Flags is something that looks very much like `Options`. If I'm not mistaken, it's an object that is passed to the `worker_threads` that takes a number of fields (with some differences) taken from `Options`.

File: lib/app-wrapper.js (hunk 1 above)
Comment (lishaduck) [reply lib/app-wrapper.js]: > soon

Today!
I'm currently cleaning up this branch locally, I should have something up today or tomorrow (certainly tomorrow in Europe. How's the future? 😉).

**Comment by lishaduck** [comment]
Yikes! I'm up to 17 commits on my branch. I really need to split them up 😬

**Pull request closed without merging**